### ISSUE TITLE
Couldn't establish connection between socket.io client (4.2.0) to socket server(4.2.0)

### ISSUE THREAD

**Issue by alpha-titan** [socketio/socket.io-client#1495]
There are some bugs in the `4.2.0`  client, I am not able to connect to the server.

I have **enabled** CORS on the **socket server** as well as the **express server**

I was stuck at this issue for hours tried everything no luck establish a WebSocket connection with the server, finally downgraded to `4.1.3`

To reproduce

- [x]  Create a React app with Vite js

- [x] `npm init vite@latest react_app`

- [x] install `socket.io-client` 4.2.0

- [x] Create an express server running on a different `port`

- [x] install `socket.io` 4.2.0

**Client**

Add the below code in `app.jsx`

```
const [socket, setSocket] = useState();
const handleSocket = () => {
        newSocket = io("ws://localhost:4000"); // HTTP didn't work either
	// newSocket.connect(); //? manual connection didn't work
	console.log(newSocket.connected); // false
        setSocket(newSocket); 
}
useEffect(()=>{
         handleSocket();
},[setSocket])

```
This exact code is working when downgrading the client socket version to `4.1.3`

**Server**

```
const express = require("express");
const cors = require("cors");
const http = require("http");
const { Server } = require("socket.io");
const app = express();
app.use(cors());
const httpServer = http.createServer(app);
const io = new Server(httpServer, {
	cors: {
		origin: "*",
	},
});

io.on("connection", (serverSocket) => {
	serverSocket.emit("welcome", { data: "Welcome to electron chat" });
	//? broadcast the event to the namspace
	console.log(serverSocket.id);
	io.of("/").emit("messageFromServer", {
		message: `${serverSocket.id} has joined`,
	});
	//? listen for messageFromClient event in the main namespace
	serverSocket.on("disconnect", () => {
		console.log(`${socket.id} has disconnected`);
	});
       
});

httpServer.listen(4000, () => {
	console.log("Server running on 4000");
});


```

Upon inspecting Network for ws connections

using `4.1.3` connection is created

<img width="734" alt="Screenshot 2021-09-20 at 11 11 07 PM" src="https://user-images.githubusercontent.com/31860431/134048989-499b5309-2b5c-41cf-b58e-9192ee6cc22e.png">

using `4.2.0` pending 

<img width="725" alt="Screenshot 2021-09-20 at 11 13 34 PM" src="https://user-images.githubusercontent.com/31860431/134049132-17b87186-c28e-447b-9177-6517d51e5894.png">

when I use the `CDN` and comment out the import it works again.

**Comment by darrachequesne** [comment]
I wasn't able to reproduce: https://github.com/socketio/socket.io-fiddle/tree/sample-react-app

Could you please check?

Note: could it be a problem of `integrity` [here](https://socket.io/docs/v4/client-installation/#from-a-cdn)?

**Comment by swearl** [comment]
same issue in vue 3.2

4.2 client cannot connect 4.2 server.
but 4.1.3 client can connect 4.2 server.

**Comment by darrachequesne** [comment]
It seems there is actually a problem with some bundlers (see [here](https://github.com/vitejs/vite/issues/4798#issuecomment-924620798)), I'm digging into this.

**Comment by alpha-titan** [comment]
Hi Sorry for the late reply,[here](https://github.com/alpha-titan/reatimeAPP/blob/master/videoCall/package.json )  is the link to my repo I made a sample app to reproduce the error

**Comment by darrachequesne** [comment]
@alpha-titan thanks! I've suggested a temporary workaround here: https://github.com/vitejs/vite/issues/4798#issuecomment-925689898

**Comment by alpha-titan** [comment]
Thanks @darrachequesne

**Comment by darrachequesne** [comment]
It should be fixed by [this commit](https://github.com/socketio/engine.io-client/commit/49719142f65e23efa65fca4f66765ded5d955972), included in version `4.3.x`.
